extensions/interactions/LogicProof/LogicProof.js

```diff
@@ -76,6 +92,13 @@ oppia.directive('oppiaInteractiveLogicProof', [
           editor.setOption('lineNumbers', true);
           editor.setOption('lineWrapping', true);
 
+          // The submit button exists outside of the codemirror component, but
+          // its function needs access to the 'editor' variable, so we specify
+          // the funciton here where the variable is available.
+          $element.find('button.logic-proof-submit')[0].onclick = function() {
```

Comment:
I wouldn't do this. It seems to me that this directive is tied to a specific singleton 'editor', so could that just be stored somewhere within the body of this controller?